### ISSUE TITLE
Inconsistent breakdown of imports into groups

### ISSUE THREAD

**Issue by ghost** [ennocramer/floskell#17]
`Floskell.Pretty.prettyImports` implements a grouping rule that seems to be best described as "sort alphabetically, split into groups according to part before the first dot such that `Control.Monad` and `Control.Applicative` ends in one group and `Data.Maybe` with `Data.Either` in a different group.

```haskell
{-# LANGUAGE A #-}
{-# LANGUAGE B #-}
{-# LANGUAGE C #-}
module Foo where

import Control.Applicative
import Control.Monad
import Data.Maybe
```
this ^ gets formatted as

```haskell
{-# LANGUAGE A #-}
{-# LANGUAGE B #-}
{-# LANGUAGE C #-}


module Foo where

import           Control.Applicative
import           Control.Monad

import           Data.Maybe
```

But this
```haskell
{-# LANGUAGE A #-}
{-# LANGUAGE C #-}
{-# LANGUAGE B #-}


module Foo where

import Control.Applicative
import Data.Maybe
import Control.Monad
```

gets formatted as this

```haskell
{-# LANGUAGE A #-}

{-# LANGUAGE B #-}
{-# LANGUAGE C #-}

module Foo where

import           Control.Applicative

import           Control.Monad

import           Data.Maybe
```
which is kind of strange, note all the new lines.

If it's a bug - it's caused by `lined` doing something strange, if it's a feature - it should probably be documented.

**Comment by ghost** [comment]
For imports replacing `map lined` with `map (inter newline . map pretty)` does the trick.

**Comment by ennocramer** [comment]
It's a bug, caused by the interaction of sorting and trying to preserve vertical space.  `lined` relies on source location produced by the parser to determine where extra newlines should be inserted, but this information is only correct when the ordering of elements is preserved. Currently, the sorting does not re-compute that information which causes `lined` to insert bogus empty lines.

**Comment by pacak** [comment]
But for imports and language pragmas we don't really care about preserving this vertical space, do we? Current sorting/grouping explicitly defines new set of vertical spaces - by first component of module name and for sorting/splitting of language pragmas - again, we define a new block, one pragma per line, sorted alphabetically.

**Comment by ennocramer** [comment]
It's a stop-gap measure because the defined grouping feels to simplistic. I think the whole system will be replaced by something more sophisticated before the next release. I like the basic approach of #18, but I think I want to add some more features.

**Comment by pacak** [comment]
Well, if we are talking features - I'd like to be able to render imports as something like this:

```haskell
import Control.Monad
import Data.Aeson
import Data.Aeson.Types
    ( Parser
    , parseMaybe
    )
import Data.Foldable
```

When there's several MRs changing imports of a same file this results in conflicts, making it one word per line wastes some space but simplifies conflict resolution. Currently it's hardcoded to `listAutoWrap Other "(" ")" "," imports` so this needs to be exposed to user.

And one more feature - something to format only imports and language pragmas without touching anything else. Mostly to run within CI to let the user know if he follows formatting conventions or not.

**Comment by ennocramer** [comment]
@pacak, you should be able to get the import list formatting now, using layout=vertical and indent=indent-by 4.

**Comment by ennocramer** [comment]
This should be solved with #18 and `d6498e81`.
